plugins/optimization-detective/detect.js

```diff
@@ -392,12 +396,25 @@ export default async function detect( {
 	}
 
 	// Abort if the client already submitted a URL Metric for this URL and viewport group.
-	const alreadySubmittedSessionStorageKey =
-		await getAlreadySubmittedSessionStorageKey(
-			currentETag,
-			currentUrl,
-			urlMetricGroupStatus
-		);
+	let alreadySubmittedSessionStorageKey;
+	try {
+		alreadySubmittedSessionStorageKey =
+			await getAlreadySubmittedSessionStorageKey(
+				currentETag,
+				currentUrl,
+				urlMetricGroupStatus
+			);
+	} catch ( err ) {
+		if ( err.message === 'Web Crypto API is unavailable' ) {
+			error(
+				'Unable to create session storage key: Web Crypto API is not available. This API is only available in secure contexts (HTTPS). Detection cannot proceed. If you are testing locally, ensure you use HTTPS or run on localhost.'
+			);
+		} else {
+			error( 'Unable to create session storage key: ' + err.message );
+		}
+		return;
```

Comment:
```suggestion
		alreadySubmittedSessionStorageKey =
			await getAlreadySubmittedSessionStorageKey(
				currentETag,
				currentUrl,
				urlMetricGroupStatus
			);
	} catch ( err ) {
		alreadySubmittedSessionStorageKey = 'odSubmitted';
		error( 'Unable to create session storage key. This may be due to not running on HTTPS. Error: ' + err.message );
```

Reply:
Or, instead of this, maybe this `try`/`catch` would make sense inside of the `getAlreadySubmittedSessionStorageKey` function?